pkg/runtime.v2/framework/plugins/coscheduling/indexer.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2024 The Kubeflow Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package coscheduling
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kubeflowv2 "github.com/kubeflow/training-operator/pkg/apis/kubeflow.org/v2alpha1"
+)
+
+var (
+	TrainingRuntimeContainerRuntimeClassKey        = ".trainingRuntimeSpec.jobSetTemplateSpec.replicatedJobs.podTemplateSpec.runtimeClassName"
```

Comment:
Should the indexer be equal to the Golang struct or `json` name ?
E.g. the trainingRuntimeSpec is named as `.spec`
https://github.com/kubeflow/training-operator/blob/22d5ddee92d4df9e167929550b610f7bfa1980d6/pkg/apis/kubeflow.org/v2alpha1/trainingruntime_types.go#L77

Reply:
We can specify the arbitrary key name. But the key is global within the training-ooerator.